cores/arduino/stm32/usb/usbd_desc.c

```diff
@@ -21,37 +21,68 @@
 #include "usbd_core.h"
 #include "usbd_desc.h"
 #include "utils.h"
+#include <variant.h>
 
 /* Private typedef -----------------------------------------------------------*/
 /* Private define ------------------------------------------------------------*/
-#if USBD_VID == 0x2341
+
+/* USB VID and PID: Either both or neither must be specified. If not
+ * specified, default to the ST VID, with a PID assigned to HID or a PID
+ * assigned to CDC devices. */
+#if !USBD_PID && !USBD_VID

```

Comment:
Ah, good point. It should `#undef` both macros there before redefining them.

Reply:
Yes but this is not enough.
In any case the `USBD_PID`  and `USBD_VID`  are defined...
```suggestion
#if (USBD_PID == 0) && (USBD_VID == 0)
```
I'm also wondering if this should not be managed separately.

Reply:
Also, as they are always defined the later check definition is not needed.

Reply:
> -#if !USBD_PID && !USBD_VID
> +#if (USBD_PID == 0) && (USBD_VID == 0)

These two are actually equivalent, the first uses `#if`, not `#ifdef`. Also note that when a macro is *not* defined, it expands to 0 in tests, exactly to make this kind of thing work.

> Also, as they are always defined the later check definition is not needed.

Do you mean this check?

    #if !USBD_VID || !USBD_PID

If so, then that check *is* needed, since the default vid/pid are only applied when vid *and* pid are unset (undefined or 0), since it makes no sense to use a default for either one without knowing what the other one is, since only specific combinatons are valid. So, the current code in this PR will use defaults if you specify none, but give an error when you specify *either* the vid *or* the pid.

Reply:
Arf 😝 Too tired I guess...
Right, this is OK.